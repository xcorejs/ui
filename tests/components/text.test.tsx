import 'jest-styled-components';

import { Box, Em, Link, Strikethrough, Text } from '@xcorejs/ui';
import React from 'react';
import renderer from 'react-test-renderer';

test('Text component', () => {
  const component = renderer.create(
    <Text>
      <Box>
        <Text v="strong">
          Number 15:
          {' '}
          <Text v="em">Burger king foot lettuce. </Text>
        </Text>
        The last thing you'd want in your <Link href="/">Burger King burger</Link> is someone's foot fungus. But as
        it turns out, that might be what you get. A
        <Strikethrough>
          4channer
        </Strikethrough>
        uploaded a photo anonymously to the site showcasing his feet in a plastic bin of lettuce. With the statement:
        <Em>"This is the lettuce you eat at Burger King." </Em>
        Admittedly, he had shoes on.
      </Box>

      <Box>But that's even worse.</Box>
    </Text>
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
