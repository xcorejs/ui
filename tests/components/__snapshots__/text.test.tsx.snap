// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text component 1`] = `
.c0 {
  font-size: 1.5rem;
  font-family: "rubik",-apple-system,BlinkMacSystemFont,Segoe UI,Helvetica,Arial,sans-serif,Apple Color Emoji,Segoe UI Emoji;
  font-size: 1.5rem;
  line-height: 2rem;
  -webkit-transition: color 300ms;
  transition: color 300ms;
  color: #1E3441;
}

.c1 {
  border-color: #0171b6;
  border-bottom: 1px solid transparent;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-size: 1.6rem;
  font-size: 1.6rem;
  line-height: initial;
  color: #0171b6;
  -webkit-transition: color 300ms,border 300ms;
  transition: color 300ms,border 300ms;
  cursor: pointer;
  cursor: pointer;
}

.c1:hover {
  border-bottom: 1px solid transparent;
  color: #0169a9;
}

.c1 * {
  cursor: pointer;
}

<span
  className="c0"
  color="text"
  fontFamily="text"
  fontSize="1.5rem"
>
  <div
    className=""
  >
    <strong
      className="c0"
      color="text"
      fontFamily="text"
      fontSize="1.5rem"
    >
      Number 15:
       
      <em
        className="c0"
        color="text"
        fontFamily="text"
        fontSize="1.5rem"
      >
        Burger king foot lettuce. 
      </em>
    </strong>
    The last thing you'd want in your 
    <a
      className="c1"
      color="primary"
      cursor="pointer"
      display="inline-flex"
      fontSize="1.6rem"
      href="/"
    >
      Burger King burger
    </a>
     is someone's foot fungus. But as it turns out, that might be what you get. A
    <s
      className="c0"
      color="text"
      fontFamily="text"
      fontSize="1.5rem"
    >
      4channer
    </s>
    uploaded a photo anonymously to the site showcasing his feet in a plastic bin of lettuce. With the statement:
    <em
      className="c0"
      color="text"
      fontFamily="text"
      fontSize="1.5rem"
    >
      "This is the lettuce you eat at Burger King." 
    </em>
    Admittedly, he had shoes on.
  </div>
  <div
    className=""
  >
    But that's even worse.
  </div>
</span>
`;
